service/src/main/java/org/ehrbase/dao/access/jooq/FolderAccess.java

```diff
@@ -110,70 +110,61 @@ public FolderAccess(I_DomainAccess domainAccess, UUID ehrId, I_ContributionAcces
-
-    @Override
-    public Boolean update(Timestamp transactionTime) {
-        // not valid, because missing mandatory data, like committer
-        throw new InternalServerException(CALLED_INVALID_ACCESS_LAYER_METHOD);
-    }
-
-    @Override
-    public Boolean update(final Timestamp transactionTime, final boolean force) {
-        // not valid, because missing mandatory data, like committer
-        throw new InternalServerException(CALLED_INVALID_ACCESS_LAYER_METHOD);
-    }
+    // *************Data Access and modification methods*****************
 
+    /**
+     * {@inheritDoc}
+     */
     @Override
-    public Boolean update(final Timestamp transactionTime, final boolean force, UUID contribution,
-        UUID systemId, UUID committerId, String description, ContributionChangeType changeType) {
+    public boolean update(final LocalDateTime transactionTime, UUID systemId, UUID committerId, String description, ContributionChangeType changeType) {
         /*create new contribution*/
         UUID oldContribution = this.folderRecord.getInContribution();
         UUID newContribution;
 
-        // if no custom contribution is provided create a new one, otherwise use given one
-        if (contribution == null) {
-            UUID contributionAccessEhrId = this.contributionAccess.getEhrId();
-            /*save the EHR id from oldContribution since it will be the same as this is an update operation*/
-            if (this.contributionAccess.getEhrId() == null) {
-                ContributionRecord rec = getContext().fetchOne(CONTRIBUTION, CONTRIBUTION.ID.eq(oldContribution));
-                contributionAccessEhrId = rec.getEhrId();
-            }
-            this.contributionAccess.setEhrId(contributionAccessEhrId);
-
-            this.contributionAccess.commit(transactionTime, committerId, systemId, ContributionDataType.folder, ContributionDef.ContributionState.COMPLETE, I_ConceptAccess.ContributionChangeType.MODIFICATION, description);
-            this.getFolderRecord().setInContribution(this.contributionAccess.getId());
-        } else {
-            this.getFolderRecord().setInContribution(contribution);
+        // No custom contribution is provided, so create a new one
+        UUID contributionAccessEhrId = this.contributionAccess.getEhrId();
+        /*save the EHR id from oldContribution since it will be the same as this is an update operation*/
+        if (this.contributionAccess.getEhrId() == null) {
+            ContributionRecord rec = getContext().fetchOne(CONTRIBUTION, CONTRIBUTION.ID.eq(oldContribution));
+            contributionAccessEhrId = rec.getEhrId();
         }
+        this.contributionAccess.setEhrId(contributionAccessEhrId);
+
+        this.contributionAccess.commit(Timestamp.valueOf(transactionTime), committerId, systemId,
+            ContributionDataType.folder, ContributionDef.ContributionState.COMPLETE, changeType, description);
```

Comment:
@jakesmolka I've checke the ContributionsAccess and I'm not sure I understand the role of the ContributionState, since the VERSION is the one that has a lifecycle_state, not the CONTRIBUTION, maybe it's an internal thing for the persistence to have this COMPLETE state? REF https://specifications.openehr.org/releases/RM/Release-1.1.0/common.html#_overview_5

Also if you check the code of the commit methods, internally it is using COMPLETE if the parameter is null.

Reply:
Your two newer comments are very good! I think we need to create one or two issues out of them. But again, for this PR this is unfortunately out of scope. As I stated in the beginning, most of the "added" code here is technically just moved, due to the new interfaces :slightly_smiling_face: 

Regarding your second point: True, but we agreed on not supporting states (i.e. in the direction of drafts) just yet. So the method kind of overwrites the null value. But I'm not sure what the authors intention was, why it was necessary to handle a null value and so on.

Your first point is a really good observation. You just discovered a shortcoming in the current implementation we need to address somehow. Currently it is certainly possible that the persisted data represents the state wrong, especially in cases where one contribution has many versioned objects as payload. Then each version would only have one state, the one of the linked contribution.
I'll create an issue for that.

Reply:
PS: See https://github.com/ehrbase/project_management/issues/557 and https://github.com/ehrbase/project_management/issues/556 (which was not part of your comment, but it goes in the same direction)